resolveLinkRelations.test.mjs

```diff
@@ -9,24 +11,24 @@ test("resolveLinkRelations", async (t) => {
       url: "/a.mjs",
     });
 
-    assert.ok(resolvedModules.includes("/c.mjs"));
-    assert.ok(resolvedModules.includes("/d.mjs"));
-    assert.ok(resolvedModules.includes("/lib/aa.mjs"));
-    assert.ok(resolvedModules.includes("/lib/bb.mjs"));
+    assert.ok(resolvedModules?.includes("/c.mjs"));
+    assert.ok(resolvedModules?.includes("/d.mjs"));
+    assert.ok(resolvedModules?.includes("/lib/aa.mjs"));
+    assert.ok(resolvedModules?.includes("/lib/bb.mjs"));
 
-    assert.equal(resolvedModules.length, 5);
+    assert.equal(resolvedModules?.length, 5);
 
     const resolvedModulesCached = await resolveLinkRelations({
       appPath: "test-fixtures",
       url: "/a.mjs",
     });
 
-    assert.ok(resolvedModulesCached.includes("/c.mjs"));
-    assert.ok(resolvedModulesCached.includes("/d.mjs"));
-    assert.ok(resolvedModulesCached.includes("/lib/aa.mjs"));
-    assert.ok(resolvedModulesCached.includes("/lib/bb.mjs"));
+    assert.ok(resolvedModulesCached?.includes("/c.mjs"));
+    assert.ok(resolvedModulesCached?.includes("/d.mjs"));
+    assert.ok(resolvedModulesCached?.includes("/lib/aa.mjs"));
+    assert.ok(resolvedModulesCached?.includes("/lib/bb.mjs"));
 
-    assert.equal(resolvedModulesCached.length, 5);
+    assert.equal(resolvedModulesCached?.length, 5);
```

Comment:
Type safety is pretty great, looks like it helped you notice bugs in the test code!
If you assert first that the `resolvedModulesCached` is an array, you won't have to do the `?.` from then on, which is a better approach. Also, an assertion error message in a test failure saying that the result isn't an array is easier to understand.

Reply:
Yeah that's a better idea.